crates/cargo-util/src/read2.rs

```diff
@@ -2,21 +2,28 @@ pub use self::imp::read2;
 
 #[cfg(unix)]
 mod imp {
+    use libc::{c_int, fcntl, F_GETFL, F_SETFL, O_NONBLOCK};
     use std::io;
     use std::io::prelude::*;
     use std::mem;
     use std::os::unix::prelude::*;
     use std::process::{ChildStderr, ChildStdout};
 
+    fn set_nonblock(fd: c_int) -> io::Result<()> {
+        let flags = unsafe { fcntl(fd, F_GETFL) };
```

Comment:
There is some uncertainty whether or not F_GETFL takes an argument. https://stackoverflow.com/questions/25061656/when-is-the-arg-for-f-getfl-fcntl-command-required has an interesting discussion of the issue. The [Open Group](https://pubs.opengroup.org/onlinepubs/9699919799/functions/fcntl.html) specification notes that the arg value is reserved for future growth, without specifying what that means or what it should be set to or whether or not it should be specified. [FreeBSD](https://www.freebsd.org/cgi/man.cgi?query=fcntl&sektion=2) specifically says it is ignored. Very strange.